example_scripts/login_system-cache.pwn

```diff
@@ -46,224 +39,286 @@ new Player[MAX_PLAYERS][E_PLAYERS];
 new g_MysqlRaceCheck[MAX_PLAYERS];
 
 //dialog data
-enum 
+enum
 {
-	DIALOG_INVALID,
 	DIALOG_UNUSED,
-	
+
 	DIALOG_LOGIN,
-	DIALOG_REGISTER,
+	DIALOG_REGISTER
 };
 
-
-/*
- * SA-MP callbacks
- */
+main() {}
 
 public OnGameModeInit()
 {
-	mysql_log(LOG_ERROR | LOG_WARNING, LOG_TYPE_HTML); //logs errors and warnings into a nice .html log file
-	g_SQL = mysql_connect(SQL_HOST, SQL_USER, SQL_DB, SQL_PASS);
+	new MySQLOpt: option_id = mysql_init_options();
+
+	mysql_set_option(option_id, AUTO_RECONNECT, true); // It automatically reconnects when loosing connection to mysql server
+	mysql_set_option(option_id, MULTI_STATEMENTS, true); // Allows to execute multiple queries at once
```

Comment:
There should be a comment warning about the risks of allowing multiple SQL statements: SQL injections. Usually SQL injections without allowing multiple statements are harmless. An example:
```pawn
CMD:givecookie(playerid, params[])
{
	new query[128];
	format(query, sizeof query, 
		"UPDATE players SET cookie = cookie + 1 WHERE name LIKE '%s'", params);
	mysql_tquery(g_SQL, query);
	SendClientMessage(playerid, COLOR_SUCCESS, "Cookie sent!");
	return 1;
}
```
A player calling `/givecookie %` will cause the query to give *every* player a cookie, which is not really harmful (albeit probably unfair). However, calling `/givecookie %'; DROP TABLE players; -- ` will do nothing but print an error in the MySQL log file, because multiple statements are not permitted. However, if they are, that call will __delete the whole players table__.

Yes, user input should always be escaped, but still the possible impact of having multiple statements enabled is far more greater than having them disabled.